src/status_im/transport/message/core.cljs

```diff
@@ -56,76 +31,143 @@
-                  (handle-reactions (map data-store.reactions/<-rpc reactions))))
+                  {:utils/dispatch-later [{:ms 20 :dispatch [:process-response response-js]}]}
+                  (models.reactions/receive-signal (map data-store.reactions/<-rpc reactions))))
 
       (seq invitations)
       (let [invitations (types/js->clj invitations)]
         (js-delete response-js "invitations")
         (fx/merge cofx
-                  {:utils/dispatch-later [{:ms 20 :dispatch [::process response-js]}]}
-                  (handle-invitations (map data-store.invitations/<-rpc invitations))))
+                  {:utils/dispatch-later [{:ms 20 :dispatch [:process-response response-js]}]}
+                  (models.group/handle-invitations (map data-store.invitations/<-rpc invitations))))
       (seq filters)
       (let [filters (types/js->clj filters)]
         (js-delete response-js "filters")
         (fx/merge cofx
-                  {:utils/dispatch-later [{:ms 20 :dispatch [::process response-js]}]}
-                  (handle-filters filters)))
+                  {:utils/dispatch-later [{:ms 20 :dispatch [:process-response response-js]}]}
+                  (models.filters/handle-filters filters)))
 
       (seq removed-filters)
       (let [removed-filters (types/js->clj removed-filters)]
         (js-delete response-js "removedFilters")
         (fx/merge cofx
-                  {:utils/dispatch-later [{:ms 20 :dispatch [::process response-js]}]}
-                  (handle-filters-removed filters))))))
+                  {:utils/dispatch-later [{:ms 20 :dispatch [:process-response response-js]}]}
+                  (models.filters/handle-filters-removed filters))))))
+
+(defn group-by-and-update-unviewed-counts
+  "group messages by current chat, profile updates, transactions and update unviewed counters in db for not curent chats"
+  [{:keys [current-chat-id db] :as acc} ^js message-js]
+  (let [chat-id (.-localChatId message-js)
+        message-type (.-messageType message-js)
+        from (.-from message-js)
+        new (.-new message-js)
+        current (= current-chat-id chat-id)
+        profile (models.chat/profile-chat? {:db db} chat-id)
+        tx-hash (and (.-commandParameters message-js) (.-commandParameters.transactionHash message-js))]
+
+    (cond-> acc
+      current
+      (update :messages conj message-js)
+
+      profile
+      (update :statuses conj message-js)
+
+      ;;update counter
+      (and (not current)
+           new
+           (not profile)
+           (not (= message-type constants/message-type-private-group-system-message))
+           (not (= from (multiaccounts.model/current-public-key {:db db}))))
+      (update-in [:db :chats chat-id :unviewed-messages-count] inc)
+
+      ;;conj incoming transaction for :watch-tx
+      (not (string/blank? tx-hash))
+      (update :transactions conj tx-hash)
+
+      :always
+      (update :chats conj chat-id))))
+
+(defn sort-js-messages!
```

Comment:
This is sorting incoming messages? (i.e the ones in `messages.new`)
if that's so we should just add it in status-go, as currently order is the receiving time, but if the client wants it ordered in canonical order, that's likely the place.